README.md

```diff
@@ -120,11 +159,59 @@ const fixBike = (bike: Bike, ride: Telescope<Bike>): void =>
     .evolve(fixTube);
 ```
 
-Having composition is actually a big thing which allows for: reducing testing scenarios as only small lenses need to be tested, composite lenses are as good as its elements; and decoupling responsibilities as lenses can live in the code closer to the data structures they handle.
+Having composition is actually a big thing which allows for: reducing testing scenarios as only small lenses need to be
+tested, composite lenses are as good as its elements; and decoupling responsibilities as lenses can live in the code
+closer to the data structures they handle.
+
+## Uplifting
+
+Some times you want to have a big universe represented by a type `P`, the state of your universe is an instance of that
+type. But it may happen that you also want to be in another universe `T` that shares part of the state with the `P` 
+universe but also has some stuff of its own. Like if you were modeling a city and a houses in the city, some stuff is 
+internal to the house and you don't want to keep track of it in the city.
+
+The uplift method is the Telescope version of 'What happens in Vegas stay in Vegas'. Let's suppose you have a sate type:
+
+```typescript
+interface Universe {
+  readonly planets: Planet[];
+  readonly stars: Star[];
+}
+```
+
+And you also have another type that shares the Stars part of the universe:
+
+```typescript
+interface Constellations {
+  readonly stars: Star[];
+  readonly figures: Figures[];
+}
+``` 
+
+You don't want to affect the universe when you change the _figures_ in your constellations, after all, people is always 
+looking for new forms in the stars! 
+
+The way to solve this is with `uplift`:
+
+```typescript
+const universe: Telescope<Universe>;            // Somehow you got a telescope for the universe. 
+const starLens: Lens<Universe, Constellations>; // A lens to focus on the stars.
+const initialFigures: Figures[];                // Some figures to start with;
+
+// We also need a translation between generic pairs and constellations:
+const pairToConstellations: (pair: { first: U; second: C; }) => ({stars: pair.first, figures: pair.second});
+const constellationsToPair: (cs: Constellations) => ({first: cs.stars, seccond: figures});
+
+const constellations: Telescope<Constellations> = universe
+   .magnify(starLens)
+   .uplift(initialFigures)
+   .dimap(pairToConstellations);
```

Comment:
Does `dimap` take both arguments: pairToConstellations, constellationsToPair

Reply:
If you apply `evolve` on the `uplifted` stream - would all the `telescopes` run OR only the ` .magnify(starLens)`?

Reply:
Would we only be updating the `magnifed` state? would these changes ever make it back to the `global telescope`? 

```
const abs = this.streamContract.magnify( todosLens() ).uplift([1,2,3]).dimap(pairToConstellations, <any>pairToConstellations);
    abs.stream.subscribe(console.log);

    abs.evolve(fn);
```

Reply:
Thanks, yes, `dimap` requires two functions, fixed!
Also added an explanation on which telescope will emit when.
@georgebatalinski Is the new explanation better?

Reply:
@wigahluk looks amazing - I am still trying to understand this feature - but from what i tried, it works as you described. 
The one *addition*:  you have to use `update` in order to make changes to the `uplifted` stream - if you use `evolve` it will not work?

```
    const tInt = Telescope.of(1);
    const tIntString = tInt.uplift<string>('a');
    tInt.stream.pipe(skip(1)).subscribe(next => {
      console.log('will not update');
    });
    tIntString.stream.pipe(skip(1)).subscribe(next => {
      console.log('updates: ', next);
    });
    tInt.update(2);
```


Would you be able to explain to me - how the `constrain` omits the original `this.evolver`,
hence not emitting on the main `stream` 
```
evolution => this.evolver(constrain(evolution)),
```